app/code/community/Sitewards/B2BProfessional/Model/Observer.php

```diff
@@ -251,9 +251,12 @@ protected function removePriceFilter($oBlock)
     {
         $aFilterableAttributes    = $oBlock->getData('_filterable_attributes');
         $aNewFilterableAttributes = array();
-        foreach ($aFilterableAttributes as $oFilterableAttribute) {
-            if ($oFilterableAttribute->getAttributeCode() != 'price') {
-                $aNewFilterableAttributes[] = $oFilterableAttribute;
+        
+        if (is_array($aFilterableAttributes) || is_object($aFilterableAttributes)) {
```

Comment:
Hi @andrewhowdencom, i just wanted to prevent the entries in my system.log:

> 2016-02-18T05:10:22+00:00 ERR (3): Warning: Invalid argument supplied for foreach() in /var/www/vhosts/magento_development/magento/.modman/sitewards_b2b/app/code/community/Sitewards/B2BProfessional/Model/Observer.php on line 254

Actually, i do not know which PHP function is better suitable for this, we can also use the check `is_iterable` for this.